Doc/reference/compound_stmts.rst

```diff
@@ -811,23 +834,26 @@ able to suspend execution in its *enter* and *exit* methods.
 
 The following code::
 
-    async with EXPR as VAR:
-        BLOCK
+    async with expression as target:
+        suite
 
-Is semantically equivalent to::
+is semantically equivalent to::
 
-    mgr = (EXPR)
-    aexit = type(mgr).__aexit__
-    aenter = type(mgr).__aenter__(mgr)
+    manager = (expression)
+    aexit = type(manager).__aexit__
+    value = type(manager).__aenter__(manager)
+    target = await value
+    exception = False
 
-    VAR = await aenter
     try:
-        BLOCK
+        suite
     except:
-        if not await aexit(mgr, *sys.exc_info()):
+        exception = True
+        if not await aexit(manager, *sys.exc_info()):
             raise
-    else:
-        await aexit(mgr, None, None, None)
+    finally:
```

Comment:
Why replace `else` with `finally`?

Reply:
In order to also handle the case when there is a non-local goto statement (`break`, `continue`, `return`) in the suite, cf. https://www.python.org/dev/peps/pep-0343/#specification-the-with-statement.

Reply:
@serhiy-storchaka  In order to also handle the case when there is a non-local goto statement (`break`, `continue`, `return`) in the suite, cf. https://www.python.org/dev/peps/pep-0343/#specification-the-with-statement.

Reply:
But before knowing about PEP 343 I did the same mistake: https://stackoverflow.com/questions/59322585/what-is-the-exact-try-statement-equivalent-of-the-with-statement-in-python

Reply:
@serhiy-storchaka  But before knowing about PEP 343 I did the same mistake: https://stackoverflow.com/questions/59322585/what-is-the-exact-try-statement-equivalent-of-the-with-statement-in-python